packages/server/lib/hooks/hooks.ts

```diff
@@ -76,15 +76,15 @@ export const connectionCreated = async (
 ): Promise<void> => {
     const { connection, environment, auth_mode } = createdConnectionPayload;
 
-    if (options.initiateSync === true && !isHosted) {
-        await syncManager.createSyncForConnection(connection.id as number, logContextGetter, orchestrator);
-    }
-
     if (options.runPostConnectionScript === true) {
         await postConnection(createdConnectionPayload, provider, logContextGetter);
         await externalPostConnection(createdConnectionPayload, provider, logContextGetter);
     }
 
+    if (options.initiateSync === true && !isHosted) {
```

Comment:
The previous ordering would cause a sync to fire right before the post connection script, which meant that first execution would run without the data from the post connection script.